### ISSUE TITLE
Setup auto-publishing for Safari version

### ISSUE THREAD

**Issue by fregante** [refined-github/refined-github#3686]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%24200.01%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/refined-github/issues/3686)

<details> <summary>Outdated intro</summary>

Refined GitHub is now also compatible with Safari (https://github.com/sindresorhus/refined-github/pull/3677), but it's not available on the App Store nor it's signed.

## Safari users 👋

If you're looking to use Refined GitHub for Safari, for the time being it's best to use https://github.com/lautis/refined-github-safari

</details>

## Xcode developers 👋

We need your help to:

- [ ] Configure Fastlane: https://docs.fastlane.tools/getting-started/ios/appstore-deployment/
- [ ] Setup the FastLane deployment action: https://github.com/marketplace/actions/fastlane-action

###  How to build

1. Install XCode 12 and Safari 14
2. In Safari, enable **Allow Unsigned Extensions** in the **Develop** menu.
3. Run:

	```sh
    npm install
	npm run build
	npm run pack:safari
	npm run start:safari
	```

4. In Safari, click on the extension icon while on GitHub.com to enable it.

You could also sign it locally following the instructions below: https://github.com/sindresorhus/refined-github/issues/3686#issuecomment-747455258

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $200.01)

- [<img src='https://avatars2.githubusercontent.com/u/6824881?v=4' alt='darthshadow' width=24 height=24> darthshadow](https://issuehunt.io/u/darthshadow) ($10.00)
- [<img src='https://avatars.githubusercontent.com/u/1402241?v=4' alt='fregante' width=24 height=24> fregante](https://issuehunt.io/u/fregante) ($50.01)
- [<img src='https://avatars.githubusercontent.com/u/170270?v=4' alt='sindresorhus' width=24 height=24> sindresorhus](https://issuehunt.io/u/sindresorhus) ($100.00)
- $40.00 have been anonymously funded.

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by fregante** [comment]
@lautis would you like to copy your fastlane setup to this repo?

I think that the Safari version might need some changes to `manifest.json`, so feel free to copy the `distribution` folder to `distribution-safari` in a build step if necessary.

**Comment by fregante** [comment]
@lautis would you like to copy your fastlane setup to this repo?

I think that the Safari version might need some changes to `manifest.json`, so feel free to copy the `distribution` folder to `distribution-safari` in a build step if necessary.

**Comment by 0-vortex** [comment]
can pick this up in the morning if that's ok

**Comment by fregante** [comment]
Yes please!

**Comment by 0-vortex** [comment]
Hey guys just finished reading the docs for [fastlane](https://docs.fastlane.tools/best-practices/continuous-integration/) and the great work over at [lautis/refined-github-safari/fastlane](https://github.com/lautis/refined-github-safari/compare/fastlane)

So far I have 1 issue related to certificates - looking at [docs.fastlane.tools/actions/match](https://docs.fastlane.tools/actions/match/) I believe sharing a provisioning profile would reduce the safari github action to: 

```shell
npm run build
bundle exec fastlane sign
```

Due to limitations I understand the ``appstore`` lane should be on a manual trigger to avoid potential double submit ?

If ``match`` is not appealing (overhead) the other simple solution would be to set up a CI machine with a temporary application password. I value this approach because it enables rapid development on a beta/staging lane with hardcoded `Appfile` values as described in [docs.fastlane.tools/advanced/Appfile](https://docs.fastlane.tools/advanced/Appfile/)

While the above methods are friendly enough for non-apple developers to contribute on the extension, my main motivation is keeping the workflow logic (setting up ruby, node, auth and running commands) separated from fastlane and app store flows.

If that sounds good I would be happy to move forward with a CI provisioning profile and automated github action builds for a pre-production environment - this can be used as an artifact to solve spam lagging app store releases or just compare with live version.

TL;DR: Apple release process can be hacked quickly but I am not experienced enough in solving the credentials issue :D

Happy to know your thoughts !

**Comment by gingerbeardman** [comment]
Will Safari version support macOS 10.14 Mojave? Hope so! Thanks.

**Comment by lautis** [comment]
Yes, if you have Safari 14.

**Comment by gingerbeardman** [comment]
Thank you!

**Comment by sindresorhus** [comment]
> If match is not appealing (overhead) the other simple solution would be to set up a CI machine with a temporary application password. I value this approach because it enables rapid development on a beta/staging lane with hardcoded Appfile values as described in docs.fastlane.tools/advanced/Appfile

I prefer the CI machine approach. I would like the simplest setup possible.

**Comment by gibfahn** [comment]
Got this working locally without having to tick the "Allow Unsigned Extensions" box every time you launch Safari. Basically @fregante 's instructions above, then the self-signing instructions from https://stackoverflow.com/questions/62748163/how-can-i-sign-a-safari-extension-for-just-one-computer.

```shell
git clone https://github.com/sindresorhus/refined-github && cd refined-github

# The GitHub enterprise "allow on this website" bit wasn't working in Safari, so I edited distribution/manifest.json
# To change github.com to github.mycompany.com with:
# gsed -i 's/github.com/github.mycompany.com/g' distribution/manifest.json

npm install
npm run build
npm run pack:safari
npm run start:safari

# Open app in Xcode
open "safari/Refined GitHub.xcodeproj"
```

In Xcode:
- Menu Bar -> Product -> Archive
- Click "Distribute App"
- Click "Development"
- Sign in with Apple ID and register for the (free) developer account if you haven't already
- Choose `<Me> (Personal Team)`
- Automatically manage signing
- Export
- Save to somewhere, e.g. `~/tmp`
- Close Xcode

Outside of Xcode:
- Open the app you just exported, e.g. `open ~/tmp/Refined GitHub <date> <time>/Refined GitHub.app`
- Click the button on the app when it opens to open Safari extension preferences.
- Enable app in preferences.

Then everything started working for me.

**Comment by markdorison** [comment]
To get Fastlane configured, we will need to provide Apple ID developer credentials. Is it safe to assume that the project maintainer (@sindresorhus), or a collaborator @sindresorhus trusts and specifically delegates to should handle this?

**Comment by fregante** [comment]
I just fixed some issues in the Safari version due to my dependencies, but I haven't updated them here yet.

- https://github.com/fregante/webext-options-sync-per-domain/releases/tag/v2.0.2
- https://github.com/fregante/webext-options-sync/releases/tag/v2.0.1

**Comment by L1cardo** [comment]
Any news on this issue?
I am using a self-build version, but I am looking forward to the official release.
<img width="187" alt="image" src="https://user-images.githubusercontent.com/33802186/103971480-adaa3980-51a5-11eb-83b8-9c0c4ddf4c12.png">

**Comment by sindresorhus** [comment]
Apple rejected the extension because "GitHub" is a trademarked name. We are working with GitHub on getting it approved. It will probably be another month until it's in the store though.

**Comment by gingerbeardman** [comment]
You can sometime rephrase it as "Refined for GitHub"

I had this years ago with a stats viewer for Mario Kart 8, I had to call it "Combo Explorer for Mario Kart 8"

**Comment by gingerbeardman** [comment]
> @sindresorhus you can rephrase it as "Refined for GitHub" in the App Store listing. The app file can keep the Refined GitHub name.
> 
> I had this years ago with a stats viewer for Mario Kart 8, I originally called it "Mario Kart 8 Combo Explorer" but in the end I had to call it "Combo Explorer for Mario Kart 8" under the advice of Apple App Store staff. There are lots of examples of this sort of thing on the store, search "for".
> 
> eg.
> 
> * Streaks for GitHub
>   https://apps.apple.com/gb/app/streaks-for-github/id1274680634?mt=12
> * Duplicate Cleaner for iPhoto
>   https://apps.apple.com/gb/app/duplicate-cleaner-for-iphoto/id586862299?mt=12
> * Sourcegraph for Safari
>   https://apps.apple.com/gb/app/sourcegraph-for-safari/id1543262193?mt=12
> 
> Of course there are also a bunch of apps that use GitHub in the normal way without issue!

**Comment by fregante** [comment]
Right, but we *are* getting the approval, so it’s just a matter of waiting.

**Comment by gingerbeardman** [comment]
If that's the case it wasn't clear from the Sindre's post, which was less definite.

**Comment by ogirginc** [comment]
> Right, but we are getting the approval, so it’s just a matter of waiting.

Does anyone know, how long it might take? Should I wait or just build & install myself?

**Comment by fregante** [comment]
🔔 

For the time being, everyone here is better off using https://github.com/lautis/refined-github-safari since it automatically updates via homebrew.

I also just followed @gibfahn’s instructions and created an official signed app of v21.3.5, but this lacks autoupdating so it's only good for testing. Download it from https://github.com/sindresorhus/refined-github/releases/tag/21.3.5

**Comment by gingerbeardman** [comment]
I note that this test build 21.3.5 requires 10.15 or higher. It would be good if Safari 14 on 10.14.6 could be supported, as previously mentioned: https://github.com/sindresorhus/refined-github/issues/3686#issuecomment-734803755

**Comment by gibfahn** [comment]
FWIW I've been happily using the one I built back in January without any issues. Yeah it's out of date now, but the small subset of features I actually use daily has been very solid.

**Comment by gingerbeardman** [comment]
What's the latest on the Safari Extension App Store rollout?

**Comment by sindresorhus** [comment]
I'm happy to announce that it's finally available: https://apps.apple.com/app/id1519867270

---

Backstory: GitHub has been very helpful. They got their lawyers to draft up a trademark license agreement for the name REFINED GITHUB 👍

**Comment by sindresorhus** [comment]
Please let us know if anyone else encounters this issue: https://github.com/sindresorhus/refined-github/issues/4216

**Comment by fregante** [comment]
Also question for new users: did the extension open https://github.com/sindresorhus/refined-github/issues/3543 on install?

**Comment by rakleed** [comment]
> Also question for new users: did the extension open #3543 on install?

@fregante After enabling the extension in Safari Extensions Preferences, this page does not open for me.

**Comment by gingerbeardman** [comment]
Oh no, this is 10.15 and above? 

Is there any chance to support Safari 14 on 10.14.6 as was previously promised?

It only requires a build setting change.

**Comment by j-f1** [comment]
> Also question for new users: did the extension open #3543 on install?

Nope! It put a bar on the top of my webpage asking me if I wanted to enable the extension. Clicking on the toolbar icon then crashed Safari Technology Preview. Relaunching STP, clicking the extension, and choosing “always allow for this website” enabled Refined GitHub. Perhaps the page was not opened because I’d built the Safari extension locally previously?

**Comment by yakov116** [comment]
@lautis would you be able to take a look at this?

**Comment by renatobenks** [comment]
it shouldn't be closed @fregante? 🤔

**Comment by fregante** [comment]
@renatobenks this issue is about automatic publishing. Sindre has been manually publishing the Safari releases and that's why we've been skipping versions (currently one month behind)

**Comment by L1cardo** [comment]
Why did not some one try [Fastlane](https://docs.fastlane.tools/getting-started/ios/appstore-deployment/) to solve this problem?

**Comment by fregante** [comment]
We’re just waiting for a PR 😉

**Comment by renatobenks** [comment]
Sorry @fregante, I didn't read the whole thread, so I just misunderstood that self-publishing was done alongside safari first published.

But I see it now, thanks for clarifying. I'll give it a shot for helping out!

**Comment by wearhere** [comment]
> I'm happy to announce that it's finally available: https://apps.apple.com/app/id1519867270

So the latest on RGH on Safari is that it's available 👆🏻 and fully functional, but this issue is still open because currently it has to be manually updated with the latest changes?

**Comment by yakov116** [comment]
Correct

**Comment by 0x0elliot** [comment]
hey, this look like an issue i can help out with. can you give me more context over this feature so that i can get started onto it? this seems like a simple issue from the looks of it.
